lib/fs/fs.go

```diff
@@ -359,15 +359,15 @@ func ReadFullData(r io.Reader, data []byte) error {
 
 // MustWriteData writes data to w.
 func MustWriteData(w io.Writer, data []byte) {
-	if len(data) == 0 {
-		return
-	}
-	n, err := w.Write(data)
-	if err != nil {
-		logger.Panicf("FATAL: cannot write %d bytes: %s", len(data), err)
-	}
-	if n != len(data) {
-		logger.Panicf("BUG: writer wrote %d bytes instead of %d bytes", n, len(data))
+	for len(data) != 0 {
+		n, err := w.Write(data)
+		if err != nil {
+			logger.Panicf("FATAL: cannot write %d bytes: %s", len(data), err)
+		}
+		if n == 0 {
+			logger.Panicf("BUG: wrote 0 bytes, but %d was sent", len(data))
+		}
+		data = data[n:]
```

Comment:
Actually, I'm not sure this will work, as, according to w.Write documentation, it seems it will always return non-nil err, when `n < len(data)`:
>  Write must return a non-nil error if it returns n < len(p). 

I think we should should check what type of err it returns, and if it's just incomplete write, proceed to write again the rest.

Reply:
It would be great if this issue could be reproduced in our environment. Then we could reliably detect the returned error and provide properly working workaround.

Reply:
I'm think of the way to reproduce it. What we need is to find what type of `err` is returned on partial writes, so it could be retried.